src/modules/web3/services/EthProvidersService.ts

```diff
@@ -111,4 +104,40 @@ export class EthProvidersService {
 
     return transaction;
   }
+
+  private setProviders() {
+    const supportedChainIds = Object.keys(this.appConfig.values.web3.providers);
+
+    for (const chainId of supportedChainIds) {
+      if (this.appConfig.values.web3.providers[chainId]) {
+        const provider = new ethers.providers.JsonRpcProvider(this.appConfig.values.web3.providers[chainId]);
+        this.providers[chainId] = provider;
+      }
+    }
+  }
+
+  private setSpokePoolEventQueriers() {
+    for (const chainIdStr of Object.keys(this.getProviders())) {
+      const chainId = parseInt(chainIdStr);
+      const spokePoolAddress = this.appConfig.values.web3.spokePoolContracts[chainId]?.address;
+      if (spokePoolAddress) {
+        const spokePool = SpokePool__factory.connect(
+          this.appConfig.values.web3.spokePoolContracts[chainId].address,
+          this.getProvider(chainId),
+        );
+        this.spokePoolEventQueriers[chainId] = new SpokePoolEventsQuerier(spokePool);
+      }
+    }
```

Comment:
A similar approach should be used for instantiating the `MerkleDistributorEventQuerier` objects. As a side note, `setSpokePoolEventQueriers` is not bullet proof either because it allows having defined only a single SpokePool contract per chain. Soon we'll have deployed a second SpokePool contract on mainnet. This is why an array for the MerkleDistributor contracts is better than the object that I used to define the config for the SpokePool contracts